### ISSUE TITLE
restart_necessary? should support monitoring of the swap size

### ISSUE THREAD

**Issue by jeffblake** [arches/whacamole#6]
Heroku says in their docs that swap memory can accumulate before you even hit R14's. Since swap severely degrades performance, I think it would be more prudent of whacamole to monitor swap rather than memory usage (or a combination of the two).

**Comment by jeffblake** [comment]
Checking in on this

**Comment by pomartel** [comment]
Yes, I also have this problem where the swap of my app sometimes increases long before reaching the memory limit. This greatly decreases the performance of the app and cause request timouts. If I could implement a rule like "Restart the processes when the swap reaches 60MB", it would be a lifesaver!

@jeffblake Let me know if you find the root cause for this. I read that tweaking the Ruby memory variables might help but I don't know where to start. It looks like others are also having the same problem: https://discussion.heroku.com/t/when-heroku-switch-to-use-swap-memory/778

**Comment by pomartel** [comment]
The code does not seem that complicated. I will see if I can fork this and submit a pull request.
